src/Uno.UWP/UI/ColorHelper.cs

```diff
@@ -1,20 +1,44 @@
 ﻿using System;
 using System.ComponentModel;
-using Windows.UI;
+using System.Linq;
 
 using Color = global::Windows/*Intentional space for WinUI upgrade tool*/.UI.Color;
+using KnownColors = global::Windows/*Intentional space for WinUI upgrade tool*/.UI.KnownColors;
 
 namespace Windows.UI
 {
 	public static partial class ColorHelper
 	{
-#if __ANDROID__ || __IOS__ || IS_UNIT_TESTS || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
-		[global::Uno.NotImplemented("__ANDROID__", "__IOS__", "IS_UNIT_TESTS", "__WASM__", "__SKIA__", "__NETSTD_REFERENCE__", "__MACOS__")]
+		// Uno Doc: this is a simple implementation that doesn't use localized strings
 		public static string ToDisplayName(Color color)
 		{
-			throw new global::System.NotImplementedException("The member string ColorHelper.ToDisplayName(Color color) is not implemented in Uno.");
+			// IFCPTR_RETURN(returnValue);
+			//
+			// int colorNameResourceId = 0;
+			// IFC_RETURN(ColorHelper_ToDisplayNameId(color, &colorNameResourceId));
+			// IFC_RETURN(DXamlCore::GetCurrent()->GetLocalizedResourceString(colorNameResourceId, returnValue));
+			//
+			// return S_OK;
+
+			return Enum.GetName(Enum.GetValues(typeof(KnownColors)).Cast<KnownColors>().MinBy(c => ColorDistance(new Color((uint)c), color))!);
+
+			static double ColorDistance(Color color1, Color color2)
+			{
+				// https://stackoverflow.com/a/8796867
```

Comment:
Interesting, is this algorithm not open sourced in WinUI 3?

For Avalonia I tried two things:
 1) Using the 3D Euclidian distance (geometric distance) between RGB colors. That works OK.
 2) Finally decided on doing 3D Euclidian distance in HSV color space. There is rounding included.

If the algorithm you have here works ok, great. Otherwise, feel free to use some ideas from Avalonia as well (which also implements caching).

https://github.com/AvaloniaUI/Avalonia/blob/0567a6dbe7dc122ba90e019320708e0cbc51b635/src/Avalonia.Controls.ColorPicker/Helpers/ColorHelper.cs#L63

Reply:
I'm not sure if it's somewhere in the source, but I couldn't find it. Since the original implementation also uses localization and a bunch of other stuff that I couldn't port, I thought I might as well go with something simple for the distance calculation. I tested out and I think it works pretty well, but if we find it problematic later, I will definitely take a look at your implementation. Thanks. Caching seems like it would be a definite improvement, so I will add that.

Reply:
I'll dig around in WinUI 3 code when I have a chance. This is something that could be updated later in another PR as well. I was just curious what you might have found yourself.